controller/src/main/java/io/pravega/controller/store/client/PravegaTableStoreClient.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.client;
+
+import org.apache.curator.framework.CuratorFramework;
+
+/**
+ * Pravega Table Store client.
+ */
+public class PravegaTableStoreClient implements StoreClient {
```

Comment:
This is a bit odd. It looks this is holding a curator reference for the `PravegaTables` type, but the name of the class does not really indicate it. Also, I checked the `StoreClient` interface and `getClient` is returning an `Object`! It can really be anything, so what's the expectation for this client?

Reply:
The `client` is store specific and can be null. 
It is for the store different store factories (like hostStore, CheckpointStore etc) to be aware of the client based on the store type and make use of the client accordingly.